model/task_lifecycle.go

```diff
@@ -1425,6 +1440,12 @@ func UpdateBuildAndVersionStatusForTask(t *task.Task) error {
 		return errors.Wrapf(err, "updating version '%s' status", taskVersion.Id)
 	}
 
+	if newVersionStatus != taskVersion.Status {
```

Comment:
Can we check more specifically that the old status was finished and the new status is not? Wondering if otherwise we'll hit these everytime we go from scheduled to in progress or something.